src/liballoc/str.rs

```diff
@@ -959,13 +959,15 @@ impl str {
     /// assert_eq!(s.find("Léopard"), Some(13));
     /// ```
     ///
-    /// More complex patterns with closures:
+    /// More complex patterns using point-free style and closures:
     ///
     /// ```
     /// let s = "Löwe 老虎 Léopard";
     ///
     /// assert_eq!(s.find(char::is_whitespace), Some(5));
     /// assert_eq!(s.find(char::is_lowercase), Some(1));
+    /// assert_eq!(s.find(|c: char| c.is_whitespace()), Some(5));
+    /// assert_eq!(s.find(|c: char| c.is_lowercase()), Some(1));
```

Comment:
Maybe also include a more-complex closure that can't be (easily) written point-free?  Maybe
```rust
assert_eq!(s.find(|c: char| c.is_digit(16)), Some(4));
assert_eq!(s.find(|c: char| c == 'o' || c == 'a', Some(13));
```